frontend/src/main/java/com/google/tripmeout/frontend/servlet/TripServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.tripmeout.frontend.servlet;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import com.google.tripmeout.frontend.storage.InMemoryTripModelStorage;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Singleton
+public final class TripServlet extends HttpServlet {
+  // The Trip get and delete servlet
+  private static final long serialVersionUID = 1L;
+  private final InMemoryTripModelStorage storage;
+  private static final String APPLICATION_JSON_CONTENT_TYPE = "application/json";
+  private final Gson gson;
+
+  @Inject
+  public TripServlet(final InMemoryTripModelStorage storage, Gson gson) {
```

Comment:
Why did you make these parameter final

Reply:
Well we don't want multiple instances of the storage within the Servlet so we do want assure that the InMemoryStorage is final. Someone please correct me if I'm wrong but I'm sure that if multiple instance of the storage were somehow initiated it would cause problems some how.

Reply:
Multiple instances of the in-memory storage would _definitely_ be a bad thing. I'll try to clear up the subtle misunderstanding of the current behavior and our intent here:
- `final` in this case only means that you can't reassign `storage` to point to a different value (`InMemoryTripModelStorage` in this case). Just because this variable doesn't get updated else doesn't mean that a separate storage class isn't instantiated elsewhere
  - This is why we prefer to use `final` assignments on class members (or static variables) wherever possible. Code tends to be less error-prone when values can't mutate from under you.
- The singleton nature of the storage system needs to be managed externally to this class anyway, since it's used elsewhere within the application. In the project as it stands, we're using Guice, which takes care of this for us.
  - This is why we add the singleton scope when binding a storage system in the storage module (technically still needs to be done under #81). Doing so means that only one instance is ever created and shared wherever and injected value is requested.
-  We don't actually know or care what the backing implementation of the trip storage is here. We just want something that satisfies the `TripStorage` interface, allowing us to retrieve/update trips. By taking in a `TripStorage` here instead, we can easily plug in one implementation for tests or for local server development, and a different one on the real App Engine server.
  - Plugging in different implementations in tests is critical for making our tests reliable and precise. The in-memory storage system is pretty much perfect for this scenario.